### ISSUE TITLE
Building with Clang/LLVM on Ubuntu results in an incorrect CMake environment

### ISSUE THREAD

**Issue by bitsauce** [chalet-org/chalet#195]
After the previous bug fix, I have been testing Clang (my preferred compiler) with Chalet.

Similar to last time, I'm using the SDL example, but with an updated SDL (`release-2.24.0`).

Compiling with `chalet build --toolchain gcc` works as expected, however, compiling with `chalet build --toolchain llvm` gives a strange CMake environment, and the build fails as a result.

`chalet build --toolchain gcc`:
```text
/usr/bin/cmake -G Ninja -DSDL_SHARED=OFF -DSDL_STATIC=ON -DCMAKE_C_FLAGS=-fPIC -DCMAKE_SYSTEM_NAME=Linux -DCMAKE_C_COMPILER=/usr/bin/gcc -DCMAKE_CXX_COMPILER=/usr/bin/g++ -DCMAKE_BUILD_TYPE=Release -DCMAKE_LIBRARY_ARCHITECTURE=x86_64-linux-gnu -DCMAKE_BUILD_WITH_INSTALL_RPATH=ON -S /home/marcus/chalet/chalet-example-sdl2/chalet_external/sdl2 -B /home/marcus/chalet/chalet-example-sdl2/build/x86_64-linux-gnu_Release/chalet_external/sdl2
```

`chalet build --toolchain llvm`:
```text
/usr/bin/cmake -G Ninja -DSDL_SHARED=OFF -DSDL_STATIC=ON -DCMAKE_C_FLAGS=-fPIC -DCMAKE_SYSTEM_NAME=Windows -DCMAKE_SYSTEM_PROCESSOR=x86_64 -DCMAKE_C_COMPILER=/usr/bin/clang -DCMAKE_CXX_COMPILER=/usr/bin/clang++ -DCMAKE_BUILD_TYPE=Release -DCMAKE_LIBRARY_ARCHITECTURE=x86_64-pc-linux-gnu -DCMAKE_BUILD_WITH_INSTALL_RPATH=ON -DCMAKE_LIBRARY_PATH=/usr/lib -DCMAKE_INCLUDE_PATH=/usr/include -DCMAKE_FIND_ROOT_PATH_MODE_PROGRAM=NEVER -DCMAKE_FIND_ROOT_PATH_MODE_LIBRARY=ONLY -DCMAKE_FIND_ROOT_PATH_MODE_INCLUDE=ONLY -DCMAKE_FIND_ROOT_PATH_MODE_PACKAGE=ONLY -DCMAKE_C_COMPILER_TARGET=x86_64-pc-linux-gnu -DCMAKE_CXX_COMPILER_TARGET=x86_64-pc-linux-gnu -S 
```

If you compare the two, you can see that the platform triplet is slightly different (`x86_64-linux-gnu` vs `x86_64-pc-linux-gnu`) and that the system platform is set to `Windows` with `llvm`. I was able to get the build to work by overriding the following values:

```text
CMAKE_SYSTEM_NAME=Linux
CMAKE_LIBRARY_ARCHITECTURE=x86_64-linux-gnu
CMAKE_C_COMPILER_TARGET=x86_64-linux-gnu
CMAKE_CXX_COMPILER_TARGET=x86_64-linux-gnu
```

I also notice that there are a lot more definitions in the latter build command. Does the CMake config need to be different between `gcc` and `clang` on linux, assuming we're not cross-compiling? Besides `CMAKE_C_COMPILER` and `CMAKE_CXX_COMPILER`, of course.

Thanks!

**Comment by rewrking** [comment]
I see immediately what the problem is - "pc" in the triplet was being used to switch to "Windows" cross-compile, but shouldn't be. I'll fix it.

When I was testing clang in Arch yesterday (need to install llvm in that Ubuntu VM now), I didn't have this problem because `gcc -dumpmachine` reported `x86_64-pc-linux-gnu`, the same that `clang -dumpmachine` reports, so it was not seen as a cross-compilation.

**Comment by rewrking** [comment]
Fixed in [v0.5.4](https://github.com/chalet-org/chalet/releases/tag/v0.5.4)

Just as a heads up though, if you want SDL to build correctly (or anything else, really), you'll have to use the `x86_64-linux-gnu` target triple in clang, so the command would be `chalet buildrun -t llvm-14 -a x86_64-linux-gnu` - if it doesn't match the system's gcc, the correct headers and libraries won't be found.

**Comment by bitsauce** [comment]
Nice! I tested it and it works well here. However, one issue that I have found is that it is a bit tricky to set the correct triplet via the VSCode extension. When I run `chalet buildrun -t llvm-14 -a x86_64-linux-gnu` from the command line, I get `llvm-14` as an option in VSCode, however, if I build using that option it doesn't set the correct platform triplet. Maybe this is something I can manually set in `.chaletrc`?
